library/src/main/java/com/novoda/downloadmanager/lib/DownloadBatch.java

```diff
@@ -125,4 +149,77 @@ public boolean scanCompletedMediaIfReady(DownloadScanner downloadScanner) {
         return false;
     }
 
+    public boolean shouldShowActiveItem() {
+        int visibility = info.getVisibility();
+        return visibility == NotificationVisibility.ONLY_WHEN_ACTIVE
+                || visibility == NotificationVisibility.ACTIVE_OR_COMPLETE;
+    }
+
+    public boolean shouldShowCompletedItem() {
+        int visibility = info.getVisibility();
+        return visibility == NotificationVisibility.ONLY_WHEN_COMPLETE
+                || visibility == NotificationVisibility.ACTIVE_OR_COMPLETE;
+    }
+
+    public String getBigPictureUrl() {
+        return info.getBigPictureUrl();
+    }
+
+    public String getDescription() {
+        return info.getDescription();
+    }
+
+    public String getTitle() {
+        return info.getTitle();
+    }
+
+    public long getFirstDownloadBatchId() {
+        return downloads.get(0).getId();
+    }
+
+    public Statistics getLiveStatistics(LongSparseArray<Long> downloadSpeed) {
+        long currentBytes = 0;
```

Comment:
moved it.  It's not done inside Statistics to keep the same behaviour of before i.e. this is calculated once on created and then the values are read. (rather than 3 methods that would do 3 separate loops) this was in PR #164